libsignal-service/src/models.rs

```diff
@@ -0,0 +1,33 @@
+
+/// Attachment represents an attachment received from a peer
+///
+/// Source: `textsecure/textsecure.go`
+pub struct Attachment<T> {
```

Comment:
Let's have the generic read `R` instead of `T`, because it's a reader after all.